apps/root/src/services/positionService.ts

```diff
@@ -1669,17 +1579,22 @@ export default class PositionService extends EventsManager<PositionServiceData>
         return;
       }
     }
-
     switch (transaction.type) {
       case TransactionTypes.newPosition: {
         const newPositionTypeData = transaction.typeData;
         const newId = newPositionTypeData.id;
         if (!currentPositions[`${transaction.chainId}-${newId}-v${newPositionTypeData.version}`]) {
-          currentPositions[`${transaction.chainId}-${newId}-v${newPositionTypeData.version}`] = {
-            ...currentPositions[`pending-transaction-${transaction.hash}-v${newPositionTypeData.version}`],
-            pendingTransaction: '',
+          const newPosition = getNewPositionFromTxTypeData({
+            chainId: transaction.chainId,
             id: `${transaction.chainId}-${newId}-v${newPositionTypeData.version}`,
             positionId: BigInt(newId),
+            newPositionTypeData,
+            user: transaction.from as Address,
+          });
+
+          currentPositions[`${transaction.chainId}-${newId}-v${newPositionTypeData.version}`] = {
+            ...currentPositions[`pending-transaction-${transaction.hash}-v${newPositionTypeData.version}`],
```

Comment:
This might not exist if you enter the app after the transaction is confirmed but not indexed